src/microbe_stage/CompoundCloudPlane.cs

```diff
@@ -888,15 +886,17 @@ private void PartialDiffuseEdges(int x0, int y0, int width, int height, float de
         }
     }
 
-    private void PartialAdvectCenter(int x0, int y0, int width, int height, float delta, Vector2 pos)
+    private void PartialAdvectCenter(int x0, int y0, int width, int height, float delta)
     {
+        var worldPos = ConvertToWorld(0, 0);
         for (int x = x0; x < x0 + width; ++x)
         {
             for (int y = y0; y < y0 + height; ++y)
             {
                 if (OldDensity[x, y].LengthSquared() > 1)
                 {
-                    var velocity = fluidSystem!.VelocityAt(pos + new Vector2(x, y) * Resolution) * VISCOSITY;
+                    var velocity = fluidSystem!.VelocityAt(
+                        new Vector2(worldPos.X + x * Resolution, worldPos.Z + y * Resolution)) * VISCOSITY;
```

Comment:
Sorry to nitpick, but as the clouds is one of the most performance critical systems, I think the Resolution should be read to a local variable outside the loop, this way the property (reading of which emits a method call operation) is read only once. (same in the below diff block).